test/uk/gov/hmrc/transitmovements/controllers/MovementsControllerSpec.scala

```diff
@@ -238,6 +255,64 @@ class MovementsControllerSpec
       verify(mockMessageFactory, atLeastOnce()).create(eqTo(MessageType.DeclarationData), any(), any(), any(), any(), eqTo(MessageStatus.Processing))
     }
 
+    "must return OK if XML data extraction is successful and store in object store" in {
+
+      val tempFile = SingletonTemporaryFileCreator.create()
+      when(mockTemporaryFileCreator.create()).thenReturn(tempFile)
+
+      //checkSize
+      when(mockSmallMessageLimit.checkContentSize(any[Long])).thenReturn(true)
+
+      when(mockMovementsXmlParsingService.extractDeclarationData(any[Source[ByteString, _]]))
+        .thenReturn(departureDataEither)
+
+      when(
+        mockObjectStoreService.addMessage(any[String].asInstanceOf[MovementId], any[String].asInstanceOf[MessageId], any[Source[ByteString, _]])(
+          any[ExecutionContext],
+          any[HeaderCarrier]
+        )
+      ).thenReturn(objectSummaryEither)
+
+      when(
+        mockMovementFactory.createDeparture(
+          any[String].asInstanceOf[MovementId],
+          any[String].asInstanceOf[EORINumber],
+          any[String].asInstanceOf[MovementType],
+          any[DeclarationData],
+          any[Message],
+          any[OffsetDateTime],
+          any[OffsetDateTime]
+        )
+      )
+        .thenReturn(movement)
+
+      when(
+        mockMessageFactory.createSmallMessage(
+          any[String].asInstanceOf[MessageId],
+          any[MessageType],
+          any[OffsetDateTime],
+          any[OffsetDateTime],
+          any[Option[MessageId]],
+          any[String].asInstanceOf[ObjectStoreResourceLocation],
+          any[MessageStatus]
+        )
+      ).thenReturn(message)
```

Comment:
We've been having a few issues with `any` matchers in some of our tests -- this morning we discovered that a typo had resulted in a broken system, partially because a matcher was set to `anyString()` when it could have been set to `eqTo("common-transit-convention-traders")`

Some of these should remain `any` because they are random by nature (message ID/movement ID in your case), but some of these possibly could be matched to an exact value, such as `MessageType.DeclarationData`